diff --git a/admin_views.views_default.inc b/admin_views.views_default.inc
index f13736f..3c6e07a 100644
--- a/admin_views.views_default.inc
+++ b/admin_views.views_default.inc
@@ -20,6 +20,10 @@ function admin_views_views_default_views() {
 
   // Collect all modules providing admin_views_default.
   $modules = module_list();
+  // Place admin_views at the start of the list to allow other modules to
+  // override the default views.
+  unset($modules['admin_views']);
+  $modules = array('admin_views' => 'admin_views') + $modules;
   foreach ($modules as $module) {
     $dir = DRUPAL_ROOT . '/' . drupal_get_path('module', $module) . '/admin_views_default';
     if (is_dir($dir)) {
