diff --git a/multiple_fields_remove_button.module b/multiple_fields_remove_button.module
index 05d210d..a08e270 100644
--- a/multiple_fields_remove_button.module
+++ b/multiple_fields_remove_button.module
@@ -71,35 +71,43 @@ function multiple_fields_remove_button_js() {
  * Implements hook_field_widget_form_alter().
  */
 function multiple_fields_remove_button_field_widget_form_alter(&$element, &$form_state, $context) {
-  $field_parents = isset($element['#field_parents']) ? $element['#field_parents'] : array();
-  $field_name = isset($element['#field_name']) ? $element['#field_name'] : NULL;
-  $language = isset($element['#language']) ? $element['#language'] : LANGUAGE_NONE;
-  $delta = isset($element['#delta']) ? $element['#delta'] : 0;
-
-  // Get parent which will we use into Remove Button Element.
-  $parents = array_merge(
-    $field_parents,
-    array($field_name, $language,$delta));
-    
-    // Remove button for the following field type widgets
-    $fieldwidgets = array(
-      'multiple_selects',
-      'text_textfield',
-      'date_popup',
-      'date_text',
-      'link_field',
-      'entityreference_autocomplete',
-      'number',
-      'text_textarea',
-      'text_textarea_with_summary',
-      'addressfield_standard',
-      'location'
-      );
-    // Add extra widgets with the help of others modules.
-    drupal_alter('multiple_field_remove_button_field_widgets', $fieldwidgets);
-    
-  if ($context['field']['cardinality'] == FIELD_CARDINALITY_UNLIMITED &&
-    in_array($context['instance']['widget']['type'], $fieldwidgets)) {
+  // Remove button for the following field type widgets
+  $fieldwidgets = array(
+    'multiple_selects',
+    'text_textfield',
+    'date_popup',
+    'date_text',
+    'link_field',
+    'entityreference_autocomplete',
+    'number',
+    'text_textarea',
+    'text_textarea_with_summary',
+    'addressfield_standard',
+    'location',
+  );
+  // Add extra widgets with the help of others modules.
+  drupal_alter('multiple_field_remove_button_field_widgets', $fieldwidgets);
+
+  if ($context['field']['cardinality'] == FIELD_CARDINALITY_UNLIMITED && in_array($context['instance']['widget']['type'], $fieldwidgets)) {
+
+    $original_element = $element;
+
+    if ($context['field']['type'] == 'entityreference') {
+      $original_element = $element['target_id'];
+    }
+
+    $field_parents = isset($original_element['#field_parents']) ? $original_element['#field_parents'] : array();
+    $field_name = isset($original_element['#field_name']) ? $original_element['#field_name'] : NULL;
+    $language = isset($original_element['#language']) ? $original_element['#language'] : LANGUAGE_NONE;
+    $delta = isset($original_element['#delta']) ? $original_element['#delta'] : 0;
+
+    // Get parent which will we use into Remove Button Element.
+    $parents = array_merge($field_parents, array(
+      $field_name,
+      $language,
+      $delta
+    ));
+
     $element['remove_button'] = array(
       '#delta' => $delta,
       '#name' => implode('_', $parents) . '_remove_button',
